code/controllers/subsystem/dbcore.dm

```diff
@@ -175,12 +151,34 @@ SUBSYSTEM_DEF(dbcore)
 /datum/controller/subsystem/dbcore/proc/ReportError(error)
 	last_error = error
 
-/datum/controller/subsystem/dbcore/proc/NewQuery(sql_query)
+/datum/controller/subsystem/dbcore/proc/Query(sql_query, list/params, async = TRUE)

```

Comment:
I think you might be missing a opportunity to keep both flexibility and ease of use by making this only support directly executed queries.

Also, we used queries to handle operations that have to be ran in sequence, reusing the query object also ensured it reused the same thread, ensuring the second query could not run without the first completing. You will likely find that you need to maintain this flexiability.

Two procs, one for making the query object, and one that makes a query object and just runs it, would be better here.

It will also allow you to keep async as a query property so you can do variable arguments to the query procs for the params.

Reply:
good points, i didn't realize this before.